src/React/Widgets/AnnotationStoreEditorWidget/index.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import style from 'PVWStyle/ReactWidgets/AnnotationStoreEditorWidget.mcss';
+
+import ActionListWidget from '../ActionListWidget';
+import AnnotationEditorWidget from '../AnnotationEditorWidget';
+import ToggleIconButtonWidget from '..//ToggleIconButtonWidget';
+
+function button(label, action) {
+  return <div key={label} className={style.button} onClick={action}>{label}</div>;
+}
+
+export default function annotationStoreEditorWidget(props) {
+  if (!props.annotation || !props.annotations) {
+    return null;
+  }
+
+  const listAnnotation = Object.keys(props.annotations).map((name, index) =>
+    ({
+      name,
+      action: `${index}`,
+      data: `${name}`,
+      active: (props.annotation === props.annotations[name]),
+    }));
+
+  const onActivateAnnotation = (name, action, data) => {
+    props.onChange('select', name, props.annotations[name]);
+  };
+
+  const storeAction = name => (() => props.onChange(name, props.title, props.annotation));
```

Comment:
Yes, that's fine with me.

On Tue, Aug 16, 2016 at 12:08 PM, Sebastien Jourdain <
notifications@github.com> wrote:

> In src/React/Widgets/AnnotationStoreEditorWidget/index.js
> <https://github.com/Kitware/paraviewweb/pull/205#discussion_r74966261>:
>
> > +    return null;
> > +  }
> > +
> > +  const listAnnotation = Object.keys(props.annotations).map((name, index) =>
> > +    ({
> > +      name,
> > +      action: `${index}`,
> > +      data: `${name}`,
> > +      active: (props.annotation === props.annotations[name]),
> > +    }));
> > +
> > +  const onActivateAnnotation = (name, action, data) => {
> > +    props.onChange('select', name, props.annotations[name]);
> > +  };
> > +
> > +  const storeAction = name => (() => props.onChange(name, props.title, props.annotation));
>
> // Create a storeAction callback builder
>
> Would that be enough? If not, you should come up with something that make
> more sense to you.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/paraviewweb/pull/205/files/6f1d0efd1311900d788c58696daa6dffc6e94b86#r74966261>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AS_pncKnUYoyvMCxO4JZIyY4geSQ2YLpks5qgeB_gaJpZM4Jk2I8>
> .
>